### PR TITLE
Cache block roots

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Cache block roots to speed up sync.

## Documentation
- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java [hunk 1]
```diff
@@ -146,6 +152,10 @@ public UInt64 getProposer_index() {
 
   @Override
   public Bytes32 hash_tree_root() {
+    return hashTreeRoot.get();
+  }
+
+  public Bytes32 getRoot() {
```
Comment (mbaxter) [inline ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java]:
```suggestion
  private Bytes32 calculateRoot() {
```

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java [hunk 2]
```diff
@@ -42,6 +45,9 @@
   // Body
   private BeaconBlockBody body;
```
Comment (mbaxter) [inline ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java]: Looks like we should also do this for `BlockBody`

**Review by mbaxter — APPROVED** [review]

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java (hunk 1 above)
Comment (cemozerr) [reply ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java]: Done.

File: ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java (hunk 2 above)
Comment (cemozerr) [reply ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/blocks/BeaconBlock.java]: Makes sense.

**Review by mbaxter — APPROVED** [review]

**Pull request merged**